src/module_1/module_1_meteo_api.py

```diff
@@ -9,29 +12,87 @@
 }
 VARIABLES = ["temperature_2m_mean", "precipitation_sum", "wind_speed_10m_max"]
 START_DATE = "2010-01-01"
-END_DATE = "2020-12-31"
+END_DATE = "2020-01-02"
+
+
+def get_data_from_api(url: str, params: dict[str, Any]) -> Any:
+    r = requests.get(url, params=params)
+    if r.status_code == 200:
+        try:
+            return r.json()
+        except ValueError:
+            raise Exception("Error: Could not decode JSON response")
+    else:
+        raise Exception(f"Error{r.status_code}: {r.text}")
+
+
+def process_data(data: Any, city: str) -> pd.DataFrame:
+    if "daily" in data:
+        daily_data = data["daily"]
+
+        # create a DataFrame from the daily data
```

Comment:
Same with these comments. You can avoid them. It is obvious from the code and the namings.